helix-core/src/test/java/org/apache/helix/manager/zk/TestZkBucketDataAccessor.java

```diff
@@ -59,7 +60,7 @@ public class TestZkBucketDataAccessor extends ZkTestBase {
   private final ZNRecord record = new ZNRecord(NAME_KEY);
 
   private HelixZkClient _zkClient;
-  private BucketDataAccessor _bucketDataAccessor;
+  private ZkBucketDataAccessor _bucketDataAccessor;
```

Comment:
Why not using interface but particular impl?

Reply:
I think this was because the overloaded doUpdate fn with additional params is not part of the interface, hence code throws compilation error when tried to access using BucketDataAccessor obj.

Reply:
Creating an reference to the base class and pointing to the actual object is the most common use case. I would suggest let's test that use case in unit test. If we need to access a particular API, we can do casting.

Reply:
@xyuanlu could you help explain whats the actual change to be made here. I might not have understood your comment fully.

Reply:
reverted the change.